### PR TITLE
Revises the Indexer to improve failure handling and logging (#502).

### PR DESCRIPTION
**Pull request by bwatson78**
- .gitignore: ignores any new log file posted into root folder.
- app/loggers/oai_mmsid_logger.rb and app/loggers/blackcat_logger.rb: removes these loggers in favor of Rails built-in class.
- app/services/oai_processing_service.rb: 
     - completely refactors the long processing method into smaller method calls.
     - removes the skip to another processing method that was there whenever we had two indexers going.
     - uses the Rails Logger to announce any errors or pertinent information for debugging.
     - escapes the method (and therefore rake task) if errors occur during API calls.
     - now uses RSolr to post ids needed for deletion.
     - switches to Faraday to communicate with OAI service.
- lib/marc_indexer.rb: makes Traject's default SolrWriter stop once an error is detected. Also increases the http timeout talking with Solr to 4 minutes.
- lib/tasks/marc_index_ingester.rake: institutes a Rails Logger and swaps it for the STDOUT.
- spec/*: swaps to the Rails Logger and removes expectation for the method skip.

### CODE REVIEW

**Comment by bwatson78** [comment]
No screenshots necessary.

**Review by rotated8 — CHANGES_REQUESTED** [review]
There are a couple places where it feels like we could improve performance by reducing the number of method calls we make. Can you take a look and advise?

File: app/services/oai_processing_service.rb [hunk 1]
```diff
@@ -95,10 +63,69 @@ def self.oai_to_marc
...
+    ids
+  end
+
+  def self.active_ids_xpath
+    '/oai:OAI-PMH/oai:ListRecords/oai:record/oai:metadata/marc:record/marc:controlfield[@tag="001"]'
+  end
+
+  def self.solr
+    RSolr.connect(url: ENV['SOLR_URL'], update_format: :xml, retry_503: 5, retry_after_limit: 5)
+  end
```
[the remark this answers is not in the document]
Comment (bwatson78) [reply app/services/oai_processing_service.rb:103]: Will do.

File: app/services/oai_processing_service.rb [hunk 2]
```diff
@@ -95,10 +63,69 @@ def self.oai_to_marc
...
-    time = time.strftime("%Y-%m-%d %H:%M:%S")
-    puts "#{time} - #{msg}"
-    true
+  def self.marc_url
+    { 'marc' => "http://www.loc.gov/MARC21/slim" }
+  end
+
+  def self.oai_url
+    { 'oai' => 'http://www.openarchives.org/OAI/2.0/' }
+  end
```
[the remark this answers is not in the document]
Comment (bwatson78) [reply app/services/oai_processing_service.rb:72]: I agree, they should be constants.

[the remark this answers is not in the document]
Comment (bwatson78) [reply app/services/oai_processing_service.rb]: Change made.

**Comment by bwatson78** [comment]
@rotated8 All changes have been addressed.

**Pull request merged**